corehq/apps/hqcase/api/get_list.py

```diff
@@ -0,0 +1,97 @@
+import datetime
+from itertools import chain
+
+from dateutil.parser import parse
+
+from dimagi.utils.parsing import FALSE_STRINGS
+
+from corehq.apps.es import case_search
+from corehq.apps.es import cases as case_es
+
+from .core import UserError, serialize_es_case
+
+DEFAULT_PAGE_SIZE = 20
+MAX_PAGE_SIZE = 5000
+CUSTOM_PROPERTY_PREFIX = 'property.'
+
+
+def _to_boolean(val):
+    return not (val == '' or val.lower() in FALSE_STRINGS)
+
+
+def _to_int(val, param_name):
+    try:
+        return int(val)
+    except ValueError:
+        raise UserError(f"'{val}' is not a valid value for '{param_name}'")
+
+
+def _make_date_filter(date_filter, param):
+
+    def filter_fn(val):
+        try:
+            # If it's only a date, don't turn it into a datetime
+            val = datetime.datetime.strptime(val, '%Y-%m-%d').date()
+        except ValueError:
+            try:
+                val = parse(val)
+            except ValueError:
+                raise UserError(f"Cannot parse datetime '{val}'")
+        return date_filter(**{param: val})
+
+    return filter_fn
+
+
+def _to_date_filters(field, date_filter):
+    return [
+        (f'{field}.gt', _make_date_filter(date_filter, 'gt')),
+        (f'{field}.gte', _make_date_filter(date_filter, 'gte')),
+        (f'{field}.lte', _make_date_filter(date_filter, 'lte')),
+        (f'{field}.lt', _make_date_filter(date_filter, 'lt')),
+        (f'{field}.eq', _make_date_filter(date_filter, 'eq')),
```

Comment:
I was thinking about the filter name convention here (.eq, .lt etc). I'm curious how you decided to use a `.` to define the filter. We use dots elsewhere in field names like `properties.name`. I know DRF uses the double underscore for filtering: `date__gte`. I'm not suggesting we change, just interested to get some thoughts.

Reply:
I was considering these parameters in general to be of the form `key=value`, where the key can have multiple components in a drilldown fashion, so what comes after the separator belongs to the context of what comes before.

I wasn't sure how widely recognized the double underscore was as a separator, and dots seemed straightforward.  I'm definitely open to reconsidering - it's important to get this right, ideally the first time.  One possible issue with double underscores is that you can make a case property containing them, which could get confusing to guard against.  Or are you thinking about using different separators in each context?  Like `last_modified__gte` and `property.dob`?

Reply:
I was thinking of different separators in each context but I do like the dots.